### ISSUE TITLE
[P1] Fix thread_system_adapter submit_delayed to avoid std::thread

### ISSUE THREAD

**Issue by kcenon** [kcenon/network_system#273]
## Summary

Fix `thread_system_adapter::submit_delayed()` to avoid creating detached `std::thread` for delay tasks.

## Current Implementation

**File**: `src/integration/thread_system_adapter.cpp`

```cpp
// Lines 79-97
std::future<void> thread_system_pool_adapter::submit_delayed(
    std::function<void()> task,
    std::chrono::milliseconds delay
) {
    auto promise = std::make_shared<std::promise<void>>();
    auto future = promise->get_future();

    // Problem: Creates detached thread for each delayed task
    std::thread([this, task = std::move(task), delay, promise]() mutable {
        try {
            std::this_thread::sleep_for(delay);
            bool ok = pool_->submit_task([task, promise]() mutable {
                // ...
            });
        } catch (...) {
            promise->set_exception(std::current_exception());
        }
    }).detach();

    return future;
}
```

## Problem

- Creates a new detached thread for every delayed task
- Detached threads can't be properly managed or cancelled
- Resource leak potential if many delayed tasks are created
- No way to cancel pending delayed tasks

### Option A: Add delayed task support to thread_system

Extend `thread_system::thread_pool` to support scheduled/delayed tasks:

```cpp
// In thread_system - new feature
class thread_pool {
public:
    // New method for delayed execution
    bool submit_delayed_task(std::function<void()> task, 
                             std::chrono::milliseconds delay);
};
```

### Option B: Use dedicated scheduler thread in adapter

Use a single scheduler thread with priority queue (similar to basic_thread_pool):

```cpp
class thread_system_pool_adapter {
private:
    struct DelayedTask {
        std::chrono::steady_clock::time_point execute_at;
        std::function<void()> task;
    };
    
    std::priority_queue<DelayedTask> delayed_tasks_;
    std::thread scheduler_thread_;  // Single thread for all delays
    
public:
    std::future<void> submit_delayed(std::function<void()> task,
                                     std::chrono::milliseconds delay) {
        // Queue to priority queue, scheduler thread handles execution
    }
};
```

### Option C: Use ASIO timer integration

If ASIO is available, use `asio::steady_timer` for scheduling.

## thread_system Features to Potentially Add

Consider adding to thread_system for this use case:
- `thread_pool::submit_delayed()` - Built-in delayed task support
- `cancellation_token` integration for delayed tasks

## Tasks

- [ ] Evaluate options (A, B, or C)
- [ ] Implement chosen solution
- [ ] Add cancellation support for pending delayed tasks
- [ ] Add unit tests
- [ ] Update documentation

## Acceptance Criteria

- [ ] No `std::thread().detach()` in `thread_system_adapter.cpp`
- [ ] Delayed tasks can be cancelled
- [ ] Memory-safe delayed task management
- [ ] All existing tests pass

## Related

- Epic: #271
- Depends on: #272
